usecases/config/runtime/values.go

```diff
@@ -0,0 +1,108 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package runtime
+
+import (
+	"fmt"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Type represents different types that is supported in runtime configs
+type DynamicType interface {
+	int | float64 | bool | time.Duration
```

Comment:
Allowing user defined types on these primitives would be useful as well. You need to change the constrains to:
```go
type DynamicType interface {
	~int | ~float64 | ~bool | ~time.Duration
}
```

Then you can have a dynamic flag of type:
```golang
type SomeStatus int

type ExampleDynamicValueWithUnderlyingInt DynamicValue[SomeStatus]
```

currently this ^ does not compile

Reply:
good point. Added.